src/components/selector_menu/index.js

```diff
@@ -128,23 +126,30 @@ var SelectorMenu = React.createClass({
       return;
     }
 
-    var visible = _.map(fuzzy.filter(filterBy, this.state.visible), function(result) {
-      return result.string;
-    });
+    var visible = _.pluck(fuzzy.filter(filterBy, this.getOptionNames()), 'string');
 
     this.setState({
       visible: visible
     });
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    if (nextProps.optionsList.length !== this.props.optionsList.length) {
+      this.setState({ visible: [] });
+    }
+  },
+
   render: function() {
     var wrapperClass = this.state.expanded ? 'selector__wrapper expanded' : 'selector__wrapper';
     var innerClass = this.state.expanded ? 'inner-wrapper expanded' : 'inner-wrapper';
+    var visible = this.state.visible.length > 0 ? this.state.visible : this.getOptionNames();
+    var selected = this.state.selected && this.props.defaultSelection === undefined ?
```

Comment:
The form has nothing to do with this. It's about the need to set the value for the entire element externally, when it's controlled by data coming from another source, like a flux store. When the incoming data changes, the selector needs to be able to respond to that change.

Ideally, like this:
```
<SelectorMenu value={this.props.selectedUser} />
```

Right now, since it only accepts `defaultSelection`.
```
<SelectorMenu defaultSelection={this.props.selectedUser} />
```
If that you pass canonical value into it from a store, the internal state of any previous selections will take precedence over the value being passed in. So if you ever have two SelectorMenus on the page controlling the same value, they'll get out of sync as soon as one SelectorMenu updates. This can also happen if you have an alternate means of controlling the value driving the component (like the checkboxes in the filters ui).